File: src/main/java/itmo/is/project/model/module/Module.java
package itmo.is.project.model.module;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@MappedSuperclass
public abstract class Module<Blueprint extends ModuleBlueprint> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "blueprint_id", referencedColumnName = "id", nullable = false, updatable = false)
    private Blueprint blueprint;

}

File: src/main/kotlin/itmo/isproject/model/module/Module.kt
package itmo.isproject.model.module

import itmo.isproject.model.IntIdEntity
import jakarta.persistence.*

@MappedSuperclass
abstract class Module<Blueprint : ModuleBlueprint>(

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "blueprint_id", referencedColumnName = "id", nullable = false, updatable = false)
    var blueprint: Blueprint? = null

) : IntIdEntity()
